packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -68,6 +69,11 @@ export function cloudflare(pluginConfig: PluginConfig = {}): vite.Plugin {
 							}
 						: undefined,
 				builder: {
+					// Note: when running `vite build` our plugin is called once per environment, this is usually not
+					//       a problem, but if we present logs etc those would get duplicated, so in order to avoid such
+					//       duplication we set this flag, we can come up with some different solution for the duplicated
+					//       logs if this config becomes problematic for whatever reason
+					sharedConfigBuild: true,
```

Comment:
How about we only output warnings for the environment being built?
Then we wouldn't get the duplication and this wouldn't be needed.

I am concerned what effect this setting could have on other plugins or user config that expects to be run for each environment.

Reply:
that's a good point, we could do that but one very nice thing here is that we generate a single warning that combines all the non-applicable fields from all the various workers and avoids multiple separate warnings and collates all the info together nicely 😕 

Since you're worried about this Vite option I removed it and made sure that we show the warning only once by using a boolean flag: https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/112/commits/143d13f3b09fbdf4b17c90fe7dd26fb9f87ff999 please have a look and let me know if this feels better for you

@jamesopstad what do you think? are you ok with this?

Reply:
The `boolean` flag is fine with me. I think the `sharedConfigBuild` will become the default at some point in the future so we can change things then.